### ISSUE TITLE
Custom SSH Port for Proxy

### ISSUE THREAD

**Issue by potpi3** [ytti/oxidized#1537]
I'm trying to keep everything behind an ASA firewall on this network.  I have setup the asa to forward port 2222 to port 22 on a server behind the firewall as a proxy server.  I don't see a way to specify a different port for the proxy server though.  I can do it for devices that need another port but not for the proxy server itself. Am I missing something or does it not exist?  If it doesn't exist could it be added.  Thanks much for reading.

**Comment by laf** [comment]
The ssh proxy code does `-W %h:%p` so you should just be able to set `host:port`

**Comment by potpi3** [comment]
To try this out I changed in my .db file my delimiter to ";" instead of ":" so I could set a port.

All my proxies without a port work fine like this.  Setting the port doesn't seem to work.  Entry looks like this:

CISCO_3650;10.10.102.4;cisco;group1;192.168.20.20:2222


Logs show:

ssh: Could not resolve hostname 192.168.20.20:2222: Name or service not known

**Comment by potpi3** [comment]
After some more digging I see that the -W %h:%p is for the end host that you are backing up not the proxy.  So it doesn't change the port of the proxy server.

I found if I change ssh.rb:

from:
        proxy_command += "#{proxy_host} -W %h:%p"
to:
        proxy_command += "#{proxy_host} -p 2222 -W %h:%p"

This works and it jumps through my proxy server on port 2222 and backs up the device behind it.  My issue is not all the proxy servers are port 2222 so I only did this to test to see if it works.  I am not much of a programmer and have never done anything with ruby so I have no idea how to change this to a variable that is sent by the router.db file.  Anyone have any ideas on that or know of something I am missing.

Thanks much.
